File: Problem8.py
def findFirstOccurrence(nums, target):
   
    left, right = 0, len(nums) - 1
   
    firstOccurrence = -1
   
    while left <= right:
       
        middle = left + (right - left) // 2
      
        if target == nums[middle]:
            firstOccurrence = middle
            right = middle - 1
        
        elif target < nums[middle]:
            right = middle - 1
        
        else:
            left = middle + 1
    return firstOccurrence


def findLastOccurrence(nums, target):
    
    left, right = 0, len(nums) - 1
   
    lastOccurrence = -1
   
    while left <= right:
    
        middle = left + (right - left) // 2
        
        if target == nums[middle]:
            lastOccurrence = middle
            left = middle + 1
        
        elif target < nums[middle]:
            right = middle - 1
       
        else:
            left = middle + 1
    return lastOccurrence


def searchRange(nums: List[int], target: int) -> List[int]:
    return [findFirstOccurrence(nums, target), findLastOccurrence(nums, target)]

File: problem8.java
// Time Complexity :O(log(n))
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no

class Solution {
    public int[] searchRange(int[] nums, int target) {
        return new int [] {Binary_Search_First(nums,target),Binary_Search_Last(nums,target)};       
          
    }

    public int Binary_Search_First(int[]nums,int target){
        int low=0;
        int high = nums.length-1;
        int first = -1;
        while(low<=high){
            int mid = low + (high - low)/2;
            if(nums[mid] == target){
                first = mid;
                high = mid -1;
            }else if(nums[mid] < target){
                low = mid+1;
            }else{
                high = mid -1;
            }
        }
        return first;

    }
    public int Binary_Search_Last(int[]nums,int target){
        int low=0;
        int high = nums.length-1;
        int last = -1;
        while(low<=high){
            int mid = low + (high - low)/2;
            if(nums[mid] == target){
                last = mid;
                low = mid + 1;            
            }else if(nums[mid]<target){
                low = mid + 1;
            }else{
                high = mid - 1;
            }
        }
        return last;
    }
}